src/core/backend_context.cc

```diff
@@ -839,34 +945,78 @@ BackendInputCollector::Finalize()
 
   // After the above sync all the GPU->pinned copies are complete. Any
   // deferred copies of pinned->CPU can now be done.
+
```

Comment:
InputCollector is (mostly) similar to the responder so similar comments to above apply